.eslintrc.yml

```diff
@@ -0,0 +1,20 @@
+# Source: https://github.com/per1234/.github/blob/main/workflow-templates/assets/check-typescript/.eslintrc.yml
+# See: https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/README.md#configuration
+# The code style defined in this file is the official standardized style to be used in all Arduino projects and should
+# not be modified.
+
+extends:
+  - airbnb-typescript/base
+  - prettier
+plugins:
+  - "@typescript-eslint"
+parser: "@typescript-eslint/parser"
+parserOptions:
+  project:
+    - "./tsconfig.json"
```

Comment:
```suggestion
    - "./tsconfig.eslint.json"
```

The problem with setting this to `./tsconfig.json` is that file is configured to exclude the tests:

https://github.com/arduino/setup-protoc/blob/11d1666fa871de56e3f058e3503006e8bc7a08c4/tsconfig.json#L62

That makes sense for building the project, but when linting we want to lint all the TypeScript and JavaScript files of the project, including the tests.

The solution I found when setting this up in the `arduino/setup-task` repo was to do the following:

- Change `tsconfig.json` to use and `include` key instead of `exclude`, specifying the paths relevant to the build: https://github.com/arduino/setup-task/blob/3bb910ef88b9387d5a3e3a9e688094f692ec7f55/tsconfig.json#L62
- Create a dedicated [TSConfig file](https://www.typescriptlang.org/tsconfig) for ESLint: https://github.com/arduino/setup-task/blob/main/tsconfig.eslint.json
  - This file [extends `tsconfig.json`](https://github.com/arduino/setup-task/blob/3bb910ef88b9387d5a3e3a9e688094f692ec7f55/tsconfig.eslint.json#L2), so its only content is the list of additional project paths that contain JavaScript/TypeScript code: https://github.com/arduino/setup-task/blob/3bb910ef88b9387d5a3e3a9e688094f692ec7f55/tsconfig.eslint.json#L3
- Configure ESLint to use the dedicated TSConfig file for ESLint: https://github.com/arduino/setup-task/blob/3bb910ef88b9387d5a3e3a9e688094f692ec7f55/.eslintrc.yml#L12

This is a fairly standard approach. You can see here they even use it in the **typescript-eslint** project itself:

https://github.com/typescript-eslint/typescript-eslint/blob/main/tsconfig.eslint.json